src/components/data-management/ProjectDetails.jsx

```diff
@@ -341,6 +326,46 @@ const ProjectDetails = ({ width, height }) => {
     dispatch(deleteMetadataTrack(name, activeProjectUuid));
   };
 
+  const updateSpeciesOrMetadata = (value, metadataKey) => {
+    const updateObject = metadataKey === 'species' ? { species: value } : { metadata: { [metadataKey]: value } };
+    return updateObject;
+  };
+
+  const replaceEmptyCells = (value, metadataKey) => {
+    const update = updateSpeciesOrMetadata(value, metadataKey);
+
+    const updatingSpecies = (sampleUuid) => metadataKey === 'species' && !samples[sampleUuid].species;
+    const updatingMetadata = (sampleUuid) => metadataKey !== 'species'
+      && (!samples[sampleUuid].metadata[metadataKey]
+        || samples[sampleUuid].metadata[metadataKey] === defaultNA);
+
```

Comment:
why do we have that here? I thought that function `updateSpeciesOrMetadata` is taking care of this